code/game/machinery/OpTable.dm

```diff
@@ -53,75 +46,54 @@
-		return
-	if(user.buckled || user.incapacitated()) //Is the person trying to use the table incapacitated or restrained?
-		return
-	if(!ismob(O) || !iscarbon(O)) //Only Mobs and Carbons can go on this table (no syptic patches please)
-		return
-	if(!user_buckle_mob(O, user, check_loc = FALSE))
-		return
-	take_patient(O, user)
-	return TRUE
+	return take_patient(O, user)
 
-/**
-  * Updates the `patient` var to be the mob occupying the table
-  */
+/// Updates `patient` to be a carbon mob occupying the table, and returns it
 /obj/machinery/optable/proc/update_patient()
-	var/mob/living/carbon/C = locate(/mob/living/carbon, loc)
-	if(C && IS_HORIZONTAL(C))
-		patient = C
-	else
-		patient = null
+	if(patient in buckled_mobs)
+		return patient // Current patient is still here, no need to look
+
+	patient = null
+
+	if(length(buckled_mobs))
+		for(var/mob/living/carbon/C in buckled_mobs)
+			patient = C
+
+			if(length(injected_reagents))
+				to_chat(C, "<span class='danger'>You feel a series of tiny pricks!</span>")
+
+			break
+
 	if(!no_icon_updates)
-		if(C && C.pulse)
+		if(patient && patient.pulse)
 			icon_state = "table2-active"
 		else
 			icon_state = "table2-idle"
 
-/obj/machinery/optable/Crossed(atom/movable/AM, oldloc)
-	. = ..()
-	if(iscarbon(AM) && LAZYLEN(injected_reagents))
-		to_chat(AM, "<span class='danger'>You feel a series of tiny pricks!</span>")
+	return patient
+
+/obj/machinery/optable/proc/take_patient(mob/living/carbon/new_patient, mob/living/carbon/user)
+	if((!ishuman(user) && !isrobot(user)) || !istype(new_patient))
+		return
+	if(update_patient())
+		to_chat(usr, "<span class='notice'>The table is already occupied!</span>")
+		return
+
+	// Attempt to settle the patient in
+	if(!user_buckle_mob(new_patient, user, check_loc = FALSE))
+		return // User is incapacitated, patient is already buckled to something else, etc.
 
-/obj/machinery/optable/process()
 	update_patient()
-	if(LAZYLEN(injected_reagents))
-		for(var/mob/living/carbon/C in get_turf(src))
-			if(C.stat == DEAD)
-				continue
-			var/datum/reagents/R = C.reagents
-			for(var/chemical in injected_reagents)
-				R.check_and_add(chemical,reagent_target_amount,inject_amount)
+	return TRUE
 
-/obj/machinery/optable/proc/take_patient(mob/living/carbon/new_patient, mob/living/carbon/user)
-	if(new_patient == user)
-		user.visible_message("[user] climbs on [src].","You climb on [src].")
-	else
-		visible_message("<span class='alert'>[new_patient] has been laid on [src] by [user].</span>")
-	if(new_patient.s_active) //Close the container opened
-		new_patient.s_active.close(new_patient)
-	add_fingerprint(user)
+/obj/machinery/optable/process()
 	update_patient()
 
-/obj/machinery/optable/verb/climb_on()
-	set name = "Climb On Table"
-	set category = "Object"
-	set src in oview(1)
-	if(usr.stat || !iscarbon(usr) || usr.restrained() || !check_table())
+	if(!length(injected_reagents) || patient.stat == DEAD)
```

Comment:
You need to check for patient first, hence CI fail due to runtimes.

Reply:
Not sure how this snuck through, but fixed